lib/src/runner/configuration.dart

```diff
@@ -386,6 +386,31 @@ class Configuration {
     if (this == Configuration.empty) return other;
     if (other == Configuration.empty) return this;
 
+    List<String> foldTraceOnly = other._foldTraceOnly ?? _foldTraceOnly;
+    if (other._foldTraceOnly != null && _foldTraceExcept != null) {
+      foldTraceOnly = new Set.from(other._foldTraceExcept)
```

Comment:
These fields should probably be stored as sets to begin with; that'll make this logic much cleaner.